pkg/flow/deduper.go

```diff
@@ -26,6 +26,7 @@ type deduperCache struct {
 
 type entry struct {
 	key        *ebpf.BpfFlowId
+	record     *Record
```

Comment:
I was thinking about this solution to store the whole record here but it has some downsides / challenges:

- Records are now kept in memory in two places: in the `Accounter` and now here - with different eviction timeouts, the one defined here being bigger for good reasons. So we can expect an increased memory usage as those records won't be garbage-collected until they expire from this cache.
- If a record was flushed out from `Accounter` and still present here while you revert its Duplicate flag with `fEntry.record.Duplicate = true`, this will actually have no effect downstream, so there are still cases where we'll have duplicates not flagged as such

Reply:
I wonder, does the concept of "Weak reference" exists in go? In order to keep this cache with weak refs of records, so that they're still gc'ed after Accounter flush ?

Reply:
so we are storing pointer to the record not the record itself I shared the memory usage below b4 and after and its about the same